src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -784,6 +791,14 @@ protected BoundCall MakeQueryInvocation(CSharpSyntaxNode node, BoundExpression r
 
                 receiver = new BoundBadExpression(receiver.Syntax, LookupResultKind.NotAValue, ImmutableArray<Symbol?>.Empty, ImmutableArray.Create(receiver), CreateErrorType());
             }
+            else
+            {
+                var checkedUltimateReceiver = CheckValue(ultimateReceiver, BindValueKind.RValue, diagnostics);
```

Comment:
>CheckValue [](start = 46, length = 10)

It feels like this could be premature for a BoundKind.TypeOrValueExpression receiver. Need to make sure we have tests for a "Color Color" case, covering both ways of resolving the receiver.